### PR TITLE
fix (host): fix hosts listing

### PR DESCRIPTION
**Pull request by callapa**
## Description

Fix hosts listing.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/core/pathway/pathway.php [hunk 1]
```diff
@@ -144,7 +144,7 @@
 
     if (isset($_GET["host_id"])) {
         echo '<span class="pathWayBracket" > &nbsp;&#62;&nbsp; </span>';
-        echo getMyHostName(htmlentities($_GET["host_id"], ENT_QUOTES, "UTF-8"));
+        echo htmlentities(getMyHostName((int) $_GET["host_id"]));
```
Comment (Drago37) [inline centreon/www/include/core/pathway/pathway.php:147]: why htmlentities ? Because int cast is sufficient and htmlentities is useless.

**Review by Drago37 — CHANGES_REQUESTED** [review]

File: centreon/www/include/core/pathway/pathway.php (hunk 1 above)
Comment (callapa) [reply centreon/www/include/core/pathway/pathway.php:147]: the int method to cast the host ID and htmlentities to encode all characters of the host name found before displaying it in the form.

Comment (Drago37) [reply centreon/www/include/core/pathway/pathway.php:147]: Sorry, I missread 😄 I suggest using htmlspecialchars for more security like this suggestion :  
```suggestion
        echo htmlspecialchars(getMyHostName((int) $_GET["host_id"]), ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8');
```

Comment (Drago37) [reply centreon/www/include/core/pathway/pathway.php]: it's ok for me

**Review by Drago37 — APPROVED** [review]

**Pull request merged**